openedx/openedx-platform
# feat: ESG error handling continued

## Description

Add and extend error handling for ESG.

JIRA: https://openedx.atlassian.net/browse/AU-469
FYI: @edx/masters-devs-gta 

## Supporting information

⚠️ **I suggest reviewing with whitespace changes ignored!** There are a lot of indent changes due to blocks of code now living inside of try blocks, ignoring whitespace will make it more obvious of the _significant_ changes.

- Make internal XBlock handler data/exception handling more consistent: successful responses return data, failure responses raise exceptions, sometimes with context or special-case exception types for handling.
- Failures in an XBlock handler now return `HTTP 500: { 'error': 'ERR_INTERNAL' }` except established cases where we want to return extra data (lock/grade contest)
- Fall-through exceptions returned for uncaught and BFF-originating errors `HTTP 500: { 'error': 'ERR_UNKNOWN' }`
- Removes the `ERR_GRADE_SUBMIT` error code, returns `ERR_INTERNAL` for an ORA failure to submit grade.
- Moved ORA xblock handlers to a new file.
- Refactored some test logic.
- Updated GitHub Action test setup to include `ora_staff_grader` app in test counts.

## Testing instructions

Use example collection to exercise BFF endpoints, verify adherence to contracts. These include examples for failure states and returned exceptions.
